### PR TITLE
fix: Install pip and pip-tools in upgrade script

### PR DESCRIPTION
**Pull request by Jawayria**
Updated the upgrade target script to check the compatibility of upgraded pip and pip-tools versions. For reference, look at this https://github.com/openedx/edx-repo-health/pull/271 for new check added in edx-repo-health.

### CODE REVIEW

**Comment by mariajgrimaldi** [comment]
Hello! Can you check the failing tests? they might be related to the libraries update for docs.in

File: requirements/doc.in [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
 -r test.txt               # Core and testing dependencies for this package
 
-doc8                      # reStructuredText style checker
-edx_sphinx_theme          # edX theme for Sphinx output
-readme_renderer           # Validates README.rst for usage on PyPI
-Sphinx                    # Documentation builder
+doc8==0.10.1                      # reStructuredText style checker
+edx_sphinx_theme==3.0.0          # edX theme for Sphinx output
+readme_renderer==30.0           # Validates README.rst for usage on PyPI
+Sphinx==4.5.0                    # Documentation builder
```
Comment (mariajgrimaldi) [inline requirements/doc.in:9]: why don't you add these constraints into the constraints.txt? & justify them please

File: requirements/doc.in [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 -r test.txt               # Core and testing dependencies for this package
 
-doc8                      # reStructuredText style checker
-edx_sphinx_theme          # edX theme for Sphinx output
-readme_renderer           # Validates README.rst for usage on PyPI
-Sphinx                    # Documentation builder
+doc8==0.10.1                      # reStructuredText style checker
+edx_sphinx_theme==3.0.0          # edX theme for Sphinx output
+readme_renderer==30.0           # Validates README.rst for usage on PyPI
```
Comment (mariajgrimaldi) [inline requirements/doc.in:8]: Try with readme-renderer

**Review by mariajgrimaldi — CHANGES_REQUESTED** [review]

File: requirements/doc.in (hunk 1 above)
Comment (mariajgrimaldi) [reply requirements/doc.in:9]: This worked for me in the constraints.txt:
```
doc8==0.10.1
edx_sphinx_theme==3.0.0
Sphinx==4.5.0
```
But remember to justify the pinning. Do we need it?

Comment (Jawayria) [reply requirements/doc.in:9]: I did it to test if new package versions were a problem

**Comment by Jawayria** [comment]
Usama Sadiq will carry this forward

File: requirements/constraints.txt [hunk 3]
```diff
@@ -8,5 +8,7 @@
 # pin when possible.  Writing an issue against the offending project and
 # linking to it here is good.
 
+-c common_constraints.txt
+
 # edX opaque keys Django constraint
 django < 3.0, >= 2.2
```
Comment (mariajgrimaldi) [inline requirements/constraints.txt:14]: You should remove this constraint since it's a bit old

**Comment by mariajgrimaldi** [comment]
I think I made this work by:
```
[testenv:docs]
setenv =
    PYTHONPATH = {toxinidir}
    SPHINXOPTS = -W
whitelist_externals =
    make
    rm
    twine
deps =
    -r{toxinidir}/requirements/doc.txt
commands =
    doc8 --ignore-path docs/_build README.rst docs
    rm -f docs/openedx_events.rst
    rm -f docs/modules.rst
    # -e allows for overriding setting from the environment so that SPHINXOPTS gets picked up.
    make -e -C docs clean
    make -e -C docs html
    python setup.py sdist bdist_wheel
    twine check --strict dist/*
```
Also adding `twine` to requirements/docs.in
I used [this](https://github.com/openedx/edx-ace) repo as reference

**Comment by mariajgrimaldi** [comment]
Hello there @UsamaSadiq! We're eager to push this PR forward with the changes suggested. Is there something we can do? Let me know!

**Comment by UsamaSadiq** [comment]
Sure @mariajgrimaldi , I'll finalise the PR with suggested changes within next week and will let you know.

**Comment by mariajgrimaldi** [comment]
Hello there! I merged some changes that may fix your CI issue, so you might not need more changes.

**Comment by UsamaSadiq** [comment]
@mariajgrimaldi I don't have access to update this PR and Jawayria has changed her team so I've created a new [PR](https://github.com/openedx/openedx-events/pull/104) with all the suggested changes. 
Please close this PR and review the above mentioned PR.

**Comment by mariajgrimaldi** [comment]
Got it! @UsamaSadiq, thanks for continuing this!

**Pull request closed without merging**